Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM osrf/ros:humble-desktop-full
+
+# Install dependencies (taken from devcontainer dockerfile)
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -q && \
+    apt-get update -q && \
+    apt-get install -yq --no-install-recommends \
+    lcov \
+    curl \
+    wget \
+    python3-pip \
+    python-is-python3 \
+    python3-argcomplete \
+    python3-colcon-common-extensions \
+    python3-colcon-mixin \
+    python3-rosdep \
+    libpython3-dev \
+    python3-vcstool \
+    ros-humble-tl-expected && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create ROS workspace
+WORKDIR /ros_ws/src
+
+# Clone driver code
+RUN git clone https://github.com/bdaiinstitute/spot_ros2.git .
```

Comment:
@ryan-roche meta: if this image is to be built in some CD job, why not just assume the repository will be in context?

Reply:
To be honest, I wasn't sure if you would be ok with the idea of adding an additional CD workflow, so I wrote the Dockerfile to be able to run independently.

If you're ok with adding a CD job for it, I can refactor it to use the GitHub Actions context.

Reply:
I see. Let's see how the rest feels about a CD job then.

Reply:
Ping @khughes-bdai @amessing-bdai.

Reply:
To be clear, this CD job would involve pushing the image to something like Docker Hub? 
I am in favor of this as I think it will make this repo easier to use -- there will just be some logistics we will need to figure out about how to set up the account, etc. (I can reach out to people about this today)